tests/array_api/util_functions.py

```diff
@@ -148,3 +149,81 @@ def test_pad_error(self):
         ):
             xp.pad(a, ((1, 1)), mode="constant", constant_values=((-1, 1)))
             xp.diff(a, n=2, axis=0)
+
+    @pytest.mark.parametrize("dtype", DTYPES_WITH_BOOL)
```

Comment:
Mostly out of curiosity, is there a special meaning to doing `trapz` with a `bool` array? Other tests, didn't do anything with bool arrays so far, it seems. Why `trapz`?

Reply:
It's valid to pass a `bool` array to `trapz` and we should make sure the behavior is correct :) 
I did also add the same test case for `diff` as well.